### PR TITLE
Query for provider head/tail

### PR DESCRIPTION
**Pull request by michaelavila**
Query for provider head/tail per Steven's suggestion here: https://github.com/ipfs/go-ipfs/pull/6068#discussion_r266291056

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -114,3 +114,8 @@ require (
 	gopkg.in/gemnasium/logrus-airbrake-hook.v2 v2.1.2 // indirect
 	gotest.tools/gotestsum v0.3.3
 )
+
+replace (
+	github.com/ipfs/go-datastore => github.com/ipfs/go-datastore v0.0.2-0.20190323043649-bce485ce18d1
+	github.com/ipfs/go-ds-flatfs => github.com/ipfs/go-ds-flatfs v0.0.2-0.20190316035933-d5e3c1fa14d2
```
Comment (michaelavila) [inline go.mod]: We obviously don't want to merge with these, they are only here for CI purposes until we put out a new version of datastore.

File: provider/queue.go [hunk 2]
```diff
@@ -142,40 +142,52 @@ func (q *Queue) work() {
...
 
-	var tail uint64
-	var head uint64 = math.MaxUint64
-	for entry := range results.Next() {
-		trimmed := strings.TrimPrefix(entry.Key, "/")
-		id, err := strconv.ParseUint(trimmed, 10, 64)
-		if err != nil {
-			return 0, 0, err
-		}
+func getQueueHead(ds datastore.Datastore) (uint64, error) {
```
Comment (michaelavila) [inline provider/queue.go]: I should document these new methods. I am doing that now.

Comment (michaelavila) [reply provider/queue.go]: Actually, these methods are internal and relatively small and so I won't add any comments.

File: go.mod (hunk 1 above)
Comment (Stebalien) [reply go.mod]: Released.

**Comment by Stebalien** [comment]
FYI, a "release" is now just a `git tag -as vX.Y.Z`.

**Comment by Kubuxu** [comment]
I would recommend running `go mod tidy` before releases.

**Comment by Stebalien** [comment]
(we should add that to CI)

**Pull request merged**